src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt

```diff
@@ -86,12 +87,62 @@ class SnykLanguageClient :
     private val progressEndMsgCache: Cache<String, WorkDoneProgressEnd> =
         Caffeine.newBuilder().expireAfterWrite(10, TimeUnit.SECONDS).build()
 
+
     override fun telemetryEvent(`object`: Any?) {
         // do nothing
     }
 
-    override fun publishDiagnostics(diagnostics: PublishDiagnosticsParams?) {
-        // do nothing for now
+    override fun publishDiagnostics(diagnosticsParams: PublishDiagnosticsParams?) {
+        if (diagnosticsParams == null) {
+            return
+        }
+
+        updateCache(diagnosticsParams)
+    }
+
+    private fun updateCache(diagnosticsParams: PublishDiagnosticsParams) {
+        val filePath = diagnosticsParams.uri
+
+        try {
+            getScanPublishersFor(filePath.toVirtualFile().path).forEach { (project, scanPublisher) ->
+                val snykFile = SnykFile(project, filePath.toVirtualFile())
+                val firstElement = diagnosticsParams.diagnostics.firstOrNull()
+
+                //If the diagnostics for the file is empty, clear the cache.
+                if (firstElement == null) {
+                    scanPublisher.onPublishDiagnostics("code", snykFile, emptyList())
+                    scanPublisher.onPublishDiagnostics("oss", snykFile, emptyList())
+                    return
+                }
+
+                // We always send PublishDiagnostics for one product. So if there is an item in the array, we can deduct the product.
+                val product = firstElement.source
+
+                val issueList = diagnosticsParams.diagnostics.stream().map {
+                    Gson().fromJson(it.data.toString(), ScanIssue::class.java)
+                }.toList()
+
+                when (product) {
+                    LsProductConstants.OpenSource.value -> {
+                        scanPublisher.onPublishDiagnostics(product, snykFile, issueList)
+                    }
+
+                    LsProductConstants.Code.value -> {
+                        scanPublisher.onPublishDiagnostics(product, snykFile, issueList)
+                    }
+
+                    LsProductConstants.InfrastructureAsCode.value -> {
+                        // TODO implement
+                    }
+
+                    LsProductConstants.Container.value -> {
+                        // TODO implement
+                    }
+                }
+            }
+        } catch (e: Exception) {
+            logger.error("Error processing snyk scan", e)
```

Comment:
This looks a leftover text - should probably be updated.

Reply:
Fixed!